crates/shim-client/README.md

```diff
@@ -7,11 +7,32 @@
 
 TTRPC bindings for containerd's shim events and interfaces.
 
-## Look and feel
+## Design
+
+The `containerd-shim-client` crate provides [Protobuf](https://github.com/protocolbuffers/protobuf.git) message
+and [ttRPC](https://github.com/containerd/ttrpc.git) service definitions for the
+[Containerd shim v2](https://github.com/containerd/containerd/blob/main/runtime/v2/task/shim.proto) protocol.
+
+The message and service definitions are auto-generated from protobuf source files under `vendor/`
+by using [ttrpc-codegen](https://github.com/containerd/ttrpc-rust/tree/master/ttrpc-codegen). So please do not
+edit those auto-generated source files. If upgrading/modification is needed, please follow the steps:
+- Synchronize the latest protobuf source files from the upstream projects into directory 'vendor/'.
+- Re-generate the source files by `cargo build --features=generate_bindings`.
+- Commit the synchronized protobuf source files and auto-generated source files, keeping them in synchronization.
+
+## Usage
+Add `containerd-shim-client` as a dependency in your `Cargo.toml`
+
+```toml
+[dependencies]
+containerd-shim-client = "0.1"
+```
 
 Basic client code looks as follows:
 
 ```rust
+extern crate containerd_shim_client as client;
```

Comment:
nit: I think `extern` is no longer needed since these crates target 2018 edition: https://doc.rust-lang.org/edition-guide/rust-2018/path-changes.html#no-more-extern-crate

Reply:
Fixed, still think about rust 2015 sometimes:(